src/mlpack/core/dists/discrete_distribution.hpp

```diff
@@ -153,12 +202,18 @@ class DiscreteDistribution
    *    actually from this distribution.
    */
   void Train(const arma::mat& observations,
-             const arma::vec& probabilities);
+             const arma::mat& probabilities);
 
   //! Return the vector of probabilities.
-  const arma::vec& Probabilities() const { return probabilities; }
+  const arma::vec& Probabilities() const { return probabilities[0]; }
   //! Modify the vector of probabilities.
-  arma::vec& Probabilities() { return probabilities; }
+  arma::vec& Probabilities() { return probabilities[0]; }
+
+  //! Modify the vector Martix of probabilities.
+  const std::vector<arma::vec>& MultipleProbabilities() const { return probabilities; }
+
+  //! Return the vector Martix of probabilities.
+  std::vector<arma::vec>& MultipleProbabilities() { return probabilities; }
```

Comment:
sorry for the late reply for this comment. 

Hmmm, if so, we need to warp a lot of `arma::vec` into `std::vector` like https://github.com/mlpack/mlpack/pull/830/commits/00baf60930867dd8cdfff0f80f9d00dbeb1c2bd8#diff-ad93795d91105ba0fcc6e9815d5e9ad0R68 and some places in `hmm_impl.hpp`(sorry I forget the exactly place). 

Maybe we could keep them both and it wouldn't confuse the users, I think. :)